core/network/source.go

```diff
@@ -58,4 +80,80 @@ type ConfigSource interface {
 	// OvsManagedBridges returns the names of network interfaces that
 	// correspond to OVS-managed bridges.
 	OvsManagedBridges() (set.Strings, error)
+
+	// GetBridgePorts returns the names of network interfaces that are ports ot
+	// the bridge with the input device name.
+	GetBridgePorts(string) []string
+}
+
+// ParseInterfaceType parses the DEVTYPE attribute from the Linux kernel
+// userspace SYSFS location "<sysPath/<interfaceName>/uevent" and returns it as
+// InterfaceType. SysClassNetPath should be passed as sysPath. Returns
+// UnknownInterface if the type cannot be reliably determined for any reason.
+// Example call: network.ParseInterfaceType(network.SysClassNetPath, "br-eth1")
+// TODO (manadart 2021-02-12): As with SysClassNetPath above, specific
+// implementations should be sought for this that are OS-dependent.
+func ParseInterfaceType(sysPath, interfaceName string) InterfaceType {
+	const deviceType = "DEVTYPE="
+	location := filepath.Join(sysPath, interfaceName, "uevent")
+
+	data, err := ioutil.ReadFile(location)
```

Comment:
Is it wise to read the whole file? Instead, we could use a bufio.NewScanner and read line by line.

Reply:
The file is 2 or 3 lines, so it's no big deal.